implementations/rust/ockam/ockam_vault/src/lib.rs

```diff
@@ -46,8 +45,43 @@ mod software;
 /// Main vault types: PublicKey, Secret, SecretAttributes etc...
 mod types;
 
-pub use constants;
 pub use error::*;
 pub use software::*;
 pub use traits::*;
 pub use types::*;
+
+/// Feature set compatibility checks
+
+#[cfg(all(
+    feature = "disable_default_noise_protocol",
+    not(feature = "OCKAM_XX_25519_AES256_GCM_SHA256"),
+    not(feature = "OCKAM_XX_25519_ChaChaPolyBLAKE2s"),
+    not(feature = "OCKAM_XX_25519_AES128_GCM_SHA256")
+))]
+compile_error! {"NOISE protocol name not selected, please enable on of the following features: \"OCKAM_XX_25519_ChaChaPolyBLAKE2s\", \"OCKAM_XX_25519_AES128_GCM_SHA256\", \"OCKAM_XX_25519_AES256_GCM_SHA256\""}
+
+#[cfg(all(
+    not(feature = "disable_default_noise_protocol"),
+    any(
+        feature = "OCKAM_XX_25519_AES256_GCM_SHA256",
+        feature = "OCKAM_XX_25519_ChaChaPolyBLAKE2s",
+        feature = "OCKAM_XX_25519_AES128_GCM_SHA256"
+    )
+))]
+compile_error! {"please enable disable_default_noise_protocol feature to customize Noise protocol"}
+
+#[cfg(all(
```

Comment:
The conditions here are the same than the ones on L63. I think that the fact that exactly one protocol is enabled has to be encoded using my comment above.

Reply:
Nice catch, it looks like we need to go with your long check